### ISSUE TITLE
[Android] Improve Activity Testing to include implicit Intent attacks 

### ISSUE THREAD

**Issue by LukasMarckmiller** [OWASP/owasp-mstg#2052]
Chapter **0x04h-Testing-Code-Quality, Testing for Injection Flaws (MSTG-PLATFORM-2)** could provide a better background to intent hijacking and spoofing (see e.g https://onlinelibrary.wiley.com/doi/abs/10.1002/spe.2643).

What's missing in the MSTG is testing for a tech I read about [here](https://blog.oversecured.com/Interception-of-Android-implicit-intents/). Implicit intents are rarely discovered by automated tools but can contain some critical flaws. For example in the provided blog post an implicit intent is used to call a file picker dialog. The returned file path is not properly checked and an attacker could provide a malicious app that responds to the implicit intent with arbitrary file uri's.

**Comment by cpholguera** [comment]
Hi @LukasMarckmiller, thanks for raising the issue. We should definitely cover that. Would you like to help us by writing a short test case for it?

**Comment by cpholguera** [comment]
The PR addressing this should also cover this (now closed) issue #941:

With Android Oreo, you now have new broadcast (implicit intent) limitations, as well as limitations as a background service. This will influence how the rogue apps might be able to get to your intents.

time to update the MSTG on the android sections regarding broadcasts and (possibly) background services for platform interaction for Android:
- [ ] what should we look for with broadcasts? (What is secure/insecure?)
- [ ] what should we look for with broadcast-receivers? (What is secure/insecure?)
- [ ] what about sticky broadcasts? (till when do they work? what to look for?)
- [ ] what about intent receivers?
- [ ] when is a broadcast receiver exported or not?

Task: update section "[Testing for Sensitive Functionality Exposure Through IPC (MSTG-PLATFORM-4)](https://github.com/OWASP/owasp-mstg/blob/master/Document/0x05h-Testing-Platform-Interaction.md#testing-for-sensitive-functionality-exposure-through-ipc-mstg-platform-4)" with these changes.

**Comment by LukasMarckmiller** [comment]
@cpholguera, yes i would love to contribute and help. so should I fork the chapter and include a description of the problem aswell as coding samples? Or how should a test case look like?

**Comment by cpholguera** [comment]
That's great, thank you @LukasMarckmiller!

## Getting started

You can read how to setup your contributor environment [here](https://github.com/OWASP/owasp-mstg/blob/master/CONTRIBUTING.md#how-to-set-up-your-contributor-environment).

However, if you prefer, there's a much easier way thanks to GitHub.
- Navigate to [the chapter](https://github.com/OWASP/owasp-mstg/blob/master/Document/0x05h-Testing-Platform-Interaction.md#testing-for-injection-flaws-mstg-platform-2)
- Press `.` on your keyboard

You can use the online vscode to make & push your changes (GitHub will help you with setting up your fork and branch).

## Writing the new content

You can enhance the test case `## Testing for Injection Flaws (MSTG-PLATFORM-2)` with the new info. I'd suggest:

- **Create a new subsection** `#### Implicit Intents` under `### Overview` (at the end): here you can add any relevant theory including inline links/references to the official Android docs and other helpful blog posts.
- **Create/Enhance Static & Dynamic Analysis**:
  - create a `### Static Analysis` section and `#### Testing Implicit Intents Injection`
  - in the `### Dynamic Analysis` section please put a title to the current content (e.g. `#### Testing Content Provider Injection`) and add a new `#### Testing Implicit Intents Injection` with your content.

All please as short and concise as possible. If you have any questions please let me know here or on [Slack](https://owasp.slack.com/messages/project-mobile_omtg/details/).

**Comment by LukasMarckmiller** [comment]
Thanks for the information! I'll definitely do that :)
